### PR TITLE
Fix Wallet connection issues

### PR DESCRIPTION
**Pull request by wsdt**
## Changes

- Fixes some bugs related to chain switching especially for limited networks
- cleanup

[Bildschirmaufzeichnung vom 2023-12-19, 21:32:55.webm](https://github.com/bobanetwork/gateway/assets/28724551/370cfa11-ac23-499c-8005-b83c86bf52ee)

no worries because of the icon (I stopped the dev server when recording the video).

### CODE REVIEW

File: src/util/network/network.util.ts [hunk 1]
```diff
@@ -231,8 +232,9 @@ export const networkLimitedAvailability = (
   networkType: keyof typeof NetworkType,
   network: keyof typeof Network
 ) => {
-  return !!NetworkList[networkType]?.find((n) => n.chain === network)
-    ?.limitedAvailability
+  return !!NetworkList[networkType ?? NetworkType.TESTNET]?.find(
```
Comment (jemeza) [inline src/util/network/network.util.ts:235]: Any way you can make this more readable?

**Review by jemeza — COMMENTED** [review]
Are you going to add test?

**Review by sk-enya — COMMENTED** [review]
Can you please update the code as mention by @jemeza and please check broken e2e tests.

**Comment by alvaro-ricotta** [comment]
Also repository is out-if-date

**Review by alvaro-ricotta — APPROVED** [review]

**Pull request merged**